azure-pipelines.yml

```diff
@@ -30,12 +30,12 @@ jobs:
                 versionSpec: '$(python.version)'
                 architecture: 'x64'
 
+          # Exact spaCy pin is temporary, will be removed
           - script: |
-                pip install "spacy>=3.2.0,<3.5.0"
+                pip install "spacy==3.4.2"  
```

Comment:
Since we don't get real interaction with the requirements check in the test suite (we're not raising warnings to errors), I think it's okay to just revert this to the previous `>=3.2.0,<3.5.0`.